inboundemailactivity.php

```diff
@@ -145,6 +145,27 @@ function inboundemailactivity_civicrm_themes(&$themes) {
 
 function inboundemailactivity_civicrm_pre($op, $objectName, $id, &$params) {
   if ($objectName === 'Activity' && $op === 'create' && $params['activity_type_id'] === CRM_Core_PseudoConstant::getKey('CRM_Activity_BAO_Activity', 'activity_type_id', 'Inbound Email')) {
+    // fetch the inbound email-to-activity email address contact, if no contact found then create one and use it for 'Added by' or source_contact_id 
+    // if not found ignore and let the source contact be the From contact
+    $emailToActivityContact = civicrm_api3('MailSettings', 'get', [
+      'sequential' => 1,
+      'is_default' => 0,
+      'api.Email.get' => ['email' => "\$value.username", 'is_primary' => 1, 'sequential' => 1],
+    ]);
+    // if inbound email-to-activity email address found
+    if (!empty($emailToActivityContact['values'])) {
```

Comment:
@monishdeb doesn't this need to go after old Line 149 so that the original source contact_id is correctly linked to as the target contact?

Reply:
My bad, corrected it.